src/promptflow-azure/promptflow/azure/_storage/blob/client.py

```diff
@@ -104,10 +110,22 @@ def _get_default_datastore(
         workspace_name=workspace_name,
     )
 
-    default_datastore = ml_client.datastores.get_default()
+    default_datastore = ml_client.datastores.get_default(include_secrets=True)
     if default_datastore.type != DatastoreType.AZURE_BLOB:
         raise UserErrorException(
             message=f"Default datastore {default_datastore.name} is {default_datastore.type}, not AzureBlob."
         )
 
-    return default_datastore
+    return default_datastore, _get_datastore_credential(default_datastore, ml_client.datastores)
+
+
+def _get_datastore_credential(datastore: Datastore, operations: DatastoreOperations):
+    # Reference the similar logic in azure.ai.ml._artifact._artifact_utilities
+    # https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_artifact_utilities.py#L103
```

Comment:
What error will happen if don't have 'Microsoft.MachineLearningServices/workspaces/datastores/listSecrets/action' permission?

Comment from reference code:
:keyword credential: Local credential to use for authentication. If not provided, will try to get
        credentials from the datastore, which requires authorization to perform action
        'Microsoft.MachineLearningServices/workspaces/datastores/listSecrets/action' over target datastore.

Reply:
For such case, we will have error message like "The client 'xxx@microsoft.com' with object id 'xxx' does not have authorization to perform action 'Microsoft.MachineLearningServices/workspaces/datastores/listSecrets/action' over scope '/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.MachineLearningServices/workspaces/xxx/datastores/workspaceblobstore' or the scope is invalid. If access was recently granted, please refresh your credentials.". I think its OK.